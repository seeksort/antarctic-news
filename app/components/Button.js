import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => (
    <button className='waves-effect waves-light btn' style={{backgroundColor: props.color, margin: '2px'}} onClick={props.onClick}>
      {props.text}
    </button>
);

Button.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
};

Button.defaultProps = {
  text: 'TEXT',
  color: '#1565c0',
};

module.exports = Button;
