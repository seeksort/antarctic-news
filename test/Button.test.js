import React from 'react';
import Button from './../app/components/Button';
import {shallow} from 'enzyme';

describe('<Button />', () => {
  test('renders', () => {
    const component = shallow(<Button />);
    expect(component.hasClass('waves-effect')).toEqual(true);
  });

  test('inherits text', () => {
    const component = shallow(
      <Button text='This is a button.'></Button>
    );
    expect(component.text()).toEqual('This is a button.');
  });

  test('renders default color if no color prop', () => {
    const component = shallow(<Button />);
    expect(component.get(0).props.style).toHaveProperty('backgroundColor', '#1565c0');
  });

  test('renders color if passed as prop', () => {
    const testColor = '#495975'
    const component = shallow(<Button color={testColor} />);
    expect(component.get(0).props.style).toHaveProperty('backgroundColor', testColor);
  });
});
