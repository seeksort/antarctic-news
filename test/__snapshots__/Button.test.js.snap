// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<button
  className="waves-effect waves-light btn"
  onClick={undefined}
  style={
    Object {
      "backgroundColor": "#1565c0",
      "margin": "2px",
    }
  }
>
  This is a button.
</button>
`;
